reconcilers/reconcilers.go

```diff
@@ -250,7 +250,7 @@ func (r *ResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	r.initializeConditions(resource)
 	result, err := r.reconcile(ctx, resource)
 
-	if r.hasStatus(originalResource) {
+	if r.hasStatus(resource) {
```

Comment:
Does this work for you?

```suggestion
	if r.hasStatus(originalResource) || r.hasStatus(resource) {
```

Reply:
I had this originally then thought it was unnecessary (because units passed mostly) - I guess its possible to delete status?

Reply:
fact is this new code needs a test to evidence why it must exist, as much as the old code does.

Reply:
ended up dropping the whole if check, each facet in the block is now properly self-guarded